log_outgoing_requests/handlers.py

```diff
@@ -100,5 +102,53 @@ def emit(self, record: AnyLogRecord):
 
         OutgoingRequestsLog.objects.create(**kwargs)
 
+    def log_exception(self, record, config):
+        from django.db import transaction
+
+        from .models import OutgoingRequestsLog
+
+        scrubbed_req_headers = record.req_headers.copy()
+
+        if "Authorization" in scrubbed_req_headers:
+            scrubbed_req_headers["Authorization"] = "***hidden***"
+
+        parsed_url = urlparse(record.url)
+
+        kwargs = {
+            "url": record.url,
+            "hostname": parsed_url.netloc,
+            "params": parsed_url.params,
+            "method": record.method,
+            "timestamp": record.requested_at,
+            "trace": record.exception,
+            "req_headers": self.format_headers(scrubbed_req_headers),
+        }
+
+        if config.save_body_enabled:
+            kwargs.update(  # TODO: req_content_type?
+                {
+                    "req_body_encoding": "utf-8",
+                    "req_body": bytes(record.req_body.encode("utf-8")),
+                }
+            )
+        try:
+            with transaction.atomic():
+                OutgoingRequestsLog.objects.create(**kwargs)
+        except Exception as e:
+            logger.exception(e)
+
+    def emit(self, record: AnyLogRecord):
+        from .models import OutgoingRequestsLogConfig
+
+        config = cast(OutgoingRequestsLogConfig, OutgoingRequestsLogConfig.get_solo())
+        if not config.save_logs_enabled:
+            return
+
+        # skip requests not coming from the library requests
+        if record and is_request_log_record(record):
+            return self.log_request(record, config)
+        elif record and is_exception_log_record(record):
+            return self.log_exception(record, config)
```

Comment:
It should be possible to detect whether a log record is a requests log record or not in all cases, in the end we're the ones that are emitting the records.

I think the current checks for particular attributes are far less than ideal, and I'd rather attach some explicit metadata to the log record that informs our code that it in fact originates from us.

IMO you would then be able to get rid of this branching entirely and keep the code flatter.